sdks/java/core/src/test/java/org/apache/beam/sdk/coders/DequeCoderTest.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.coders;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.Assert.assertThat;
+
+import java.util.ArrayDeque;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Deque;
+import java.util.List;
+import org.apache.beam.sdk.testing.CoderProperties;
+import org.apache.beam.sdk.transforms.windowing.GlobalWindow;
+import org.apache.beam.sdk.util.CoderUtils;
+import org.apache.beam.sdk.values.TypeDescriptor;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Test case for {@link DequeCoder}. */
+@RunWith(JUnit4.class)
+@SuppressWarnings({
+  "nullness" // TODO(https://issues.apache.org/jira/browse/BEAM-10402)
+})
+public class DequeCoderTest {
+
+  private static final Coder<Deque<Integer>> TEST_CODER = DequeCoder.of(VarIntCoder.of());
+
+  private static final List<Deque<Integer>> TEST_VALUES =
+      Arrays.asList(
+          new ArrayDeque<>(),
+          new ArrayDeque<>(Collections.singleton(13)),
+          new ArrayDeque<>(Arrays.asList(31, -5, 83)));
```

Comment:
It would be good to have some tests where the element type is not integer. For example for the case where the inner type is arrays or some other nested structure.

Reply:
Sure, I've added the following test:
```java
  @Test
  public void encodeDequeWithList() throws Exception {
    DequeCoder<List<Long>> OBJECT_TEST_CODER = DequeCoder.of(ListCoder.of(VarLongCoder.of()));

    CoderProperties.coderDecodeEncodeContentsEqual(
        OBJECT_TEST_CODER,
        new ArrayDeque<>(
            ImmutableList.of(
                ImmutableList.of(18L, 15L), ImmutableList.of(19L, 25L), ImmutableList.of(22L))));
  }
```
Let me know if there are any other cases that you would like to see.